tools/building/createTest.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+import os
+import argparse
+import pathlib
+
+
+def is_precice_root(dir):
+    detect = [
+        "CHANGELOG.md", "CMakeLists.txt", "LICENSE", "src", "tests", "cmake"
+    ]
+    return all(map(lambda c: os.path.exists(os.path.join(dir, c)), detect))
+
+
+def find_precice_root():
+    search_depth = 4
+    current = pathlib.Path(os.path.curdir).absolute()
+    candidats = [current] + list(current.parents)[:search_depth]
+    for dir in candidats:
+        if is_precice_root(dir):
+            return dir
+    raise "Unable to find the root directory of precice"
+
+
+def patharg(arg):
+    assert (arg)
+    if " " in arg:
+        raise argparse.ArgumentTypeError(
+            "The given suite name \"{}\" cannot contain spaces.".format(arg))
+    if "." in arg:
+        raise argparse.ArgumentTypeError(
+            "The given suite name \"{}\" cannot contain the file extensions.".
+            format(arg))
+    if not arg[0].isupper():
+        raise argparse.ArgumentTypeError(
+            "The given suite name \"{}\" must use CamelCase.".format(arg))
+    return arg
+
+
+def testname(arg):
+    assert (arg)
+    if " " in arg:
+        raise argparse.ArgumentTypeError(
+            "The given test name \"{}\" cannot contain spaces.".format(arg))
+    if "." in arg:
+        raise argparse.ArgumentTypeError(
+            "The given test name \"{}\" cannot contain the file extensions.".
+            format(arg))
+    if not arg[0].isupper():
+        raise argparse.ArgumentTypeError(
+            "The given test name \"{}\" must use CamelCase.".format(arg))
+    return arg
+
+
+PRECICE_TEST_BODY = """
+{
+  PRECICE_TEST(TODO);
+  precice::SolverInterface interface(context.name, context.config(), context.rank, context.size);
+
+  std::vector<VertexID> vertexIDs;
+
+  if (context.isNamed(TODO)) {
+    auto meshID = interface.getMeshID(TODO);
+    auto dataID = interface.getDataID(TODO, meshID);
+
+    std::vector<double> coords;
+
+    interface.setMeshVertices(meshID, TODO, coords.data(), vertexIDs.data());
+  } else {
+  }
+}
+"""
+
+
+def generateTestSource(name, suite, filepath):
+    if os.path.exists(filepath):
+        raise "The test source at \"{}\" already exists.".format(filepath)
+
+    includes = [
+        "<precice/SolverInterface.hpp>", "<vector>", '"testing/Testing.hpp"'
+    ]
+    suites = ["PreciceTests"] + suite
+    space = [""]
+    lines = ["#ifndef PRECICE_NO_MPI"]
+    lines += space
+    lines += ["#include " + inc for inc in includes if inc[0] != "<"]
+    lines += space
+    lines += ["#include " + inc for inc in includes if inc[0] == "<"]
+    lines += space
+    lines += ["BOOST_AUTO_TEST_SUITE({})".format(s) for s in suites]
+    lines += ["BOOST_AUTO_TEST_CASE({})".format(name), PRECICE_TEST_BODY]
+    lines += ["BOOST_AUTO_TEST_SUITE_END() // " + s for s in suites]
+    lines += space
+    lines += ["#endif // PRECICE_NO_MPI"]
```

Comment:
I would suggest to always add a header line in autogenerated files. This helps others who see this file to know that the file is autogenerated and they know this generator can be used. Use-case: Somebody implements a new integration test and uses an existing test as a template.

A header line also helps us when reviewing to see that the test originates from the generator (and complies with the standards).

Proposed header:

For `testname.cpp`
```
/**
 * @file {{testname}}.cpp This test has been generated using our tool for generation of integration tests.
     Please use this tool, if you are creating a new integration test. Don't delete this information!
     For more information, see ADD LINK TO DOCUMENTATION
 */
```

for config.xml
```
<!-- This configuration file has been generated using our configuration tool for generation of integration tests.
     Please use this tool, if you are creating a new integration test. Don't delete this comment!
     For more information, see ADD LINK TO DOCUMENTATION -->
```

Reply:
The test itself needs to be written though. So this may actually confuse people.

This would actually be a great addition to purely generated files like the `src/sources.cmake`.

Reply:
I understand that there is no clear action here. I will mark this conversation as resolved.